runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPanel.java

```diff
@@ -410,36 +411,51 @@ public void focusLost(FocusEvent e)
 				String existing = configManager.getConfiguration(cd.getGroup().value(), cid.getItem().keyName());
 
 				Color existingColor;
-				JButton colorPickerBtn;
+				ColorJButton colorPickerBtn;
+
+				boolean alphaHidden = cid.getAlpha() == null;
 
 				if (existing == null)
 				{
 					existingColor = Color.BLACK;
-					colorPickerBtn = new JButton("Pick a color");
+					colorPickerBtn = new ColorJButton("Pick a color");
 				}
 				else
 				{
 					existingColor = ColorUtil.fromString(existing);
-					colorPickerBtn = new JButton(ColorUtil.toHexColor(existingColor).toUpperCase());
+					String colorHex = ColorUtil.toHexColor(existingColor).toUpperCase();
+					if (!alphaHidden)
+					{
+						colorHex = ColorUtil.toAlphaHexColor(existingColor).toUpperCase();
+					}
+
+					colorPickerBtn = new ColorJButton(colorHex);
 				}
 
 				colorPickerBtn.setFocusable(false);
-				colorPickerBtn.setBackground(existingColor);
+				colorPickerBtn.setColor(existingColor);
 				colorPickerBtn.addMouseListener(new MouseAdapter()
 				{
 					@Override
 					public void mouseClicked(MouseEvent e)
 					{
 						RuneliteColorPicker colorPicker = colorPickerManager.create(
 							SwingUtilities.windowForComponent(ConfigPanel.this),
-							colorPickerBtn.getBackground(),
+							colorPickerBtn.getColor(),
 							cid.getItem().name(),
-							cid.getAlpha() == null);
+							alphaHidden);
 						colorPicker.setLocation(getLocationOnScreen());
 						colorPicker.setOnColorChange(c ->
 						{
-							colorPickerBtn.setBackground(c);
-							colorPickerBtn.setText(ColorUtil.toHexColor(c).toUpperCase());
+							colorPickerBtn.setColor(c);
+							if (alphaHidden)
+							{
+								colorPickerBtn.setText(ColorUtil.toHexColor(c).toUpperCase());
+							}
+							else
+							{
+								colorPickerBtn.setText(ColorUtil.toAlphaHexColor(c).toUpperCase());
```

Comment:
Internally all of the config Colors always have alpha=0xff, even when the picker doesn't have alpha enabled, so I think this can just be `colorPickerBtn.setText(ColorUtil.toAlphaHexColor(c).toUpperCase());`

Reply:
Currently I built the logic to only show the 8-digit hex codes (#aarrggbb) on config options that actually support opacity, the other ones keep the 6-digit version (#rrggbb). Using only `.toAlphaHexColor` here would cause those buttons to also have the long hex codes after they have been changed, which would be quite confusing IMO.